src/components/Api.js

```diff
@@ -0,0 +1,122 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._headers = headers;
+    this._baseUrl = baseUrl;
+  }
+
+  _processResponse = (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  };
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._processResponse);
+  }
+
+  // other methods for working with the API
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, { headers: this._headers }).then(
+      (res) => {
```

Comment:
Needs correction: You should use `this. _processResponse` method for processing the response everywhere

Reply:
When I replace promise (if (res.ok) {
          return res.json();
        } else {
          return Promise.reject(`Error: ${res.status}`);
        }) in all the other api methods with this._processResponse, I get errors which I am confused about because this._processResponse is the exact same thing just simplified in 1 name.

Reply:
Try to use this `return fetch(`${this._baseUrl}/users/me`, { headers: this._headers }).then(this. _processResponse)`